### PR TITLE
new: ping reports

### PR DESCRIPTION
**Pull request by sibicramesh**

### CODE REVIEW

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "2992af4a080708efe3f3ff79eafcfe1763c548b2"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "da6a966bf445baa42a0c469318f4b7db3218157f"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "90e46493079974dae53795f0fa8d64f6dee79534"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "fe7d54eedad5df8807bc061d5b9625287a69ffdb"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "3f8547a748ea24aff95b3dc5b2ff03379fdd6a61"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "3f8547a748ea24aff95b3dc5b2ff03379fdd6a61"
  },
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "4c139c0b53dfee0ad98d434a10edc3bd29189386"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "bfbe325d4e058ca0c626e8426bfeadc5e17634d2"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "90e46493079974dae53795f0fa8d64f6dee79534"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "bcaeba8e254833edae0fd8060ff977aec237f243"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "4c139c0b53dfee0ad98d434a10edc3bd29189386"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "a33a1efc51e84eb3b971fb124ee463d0a6d3f86e"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "90e46493079974dae53795f0fa8d64f6dee79534"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "bcaeba8e254833edae0fd8060ff977aec237f243"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "3f8547a748ea24aff95b3dc5b2ff03379fdd6a61"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "400ca4554ce09466358954ea1e4008379d0269e2"
  },
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "3b321008c7a8521a95eb073d4ed91056ea42082a"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "8947f6ed048da5524e9409ae2b6151f119050708"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "5d78ea3dc0a01ec77780a5f34abcb19352f6ae51"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "bcaeba8e254833edae0fd8060ff977aec237f243"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "5d78ea3dc0a01ec77780a5f34abcb19352f6ae51"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "bcaeba8e254833edae0fd8060ff977aec237f243"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "400ca4554ce09466358954ea1e4008379d0269e2"
  },
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "3b321008c7a8521a95eb073d4ed91056ea42082a"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "ac6ea2fe34bb114d0629bc3062bb09048129189b"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "6985fb0769f92103b1c5edef674dfc4704435b74"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "600cf7133eddb41294f44c394d9a0c8e536da229"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "400ca4554ce09466358954ea1e4008379d0269e2"
  },
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "1f7e0acc5590137621cf7888539286343b902a87"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "35784888654ca1099978fa91a3e62c1f6aa3e81c"
  }
]
```

File: specs/pingreport.spec [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    description: ID unique to a single origin and reply report.
+    type: string
+    exposed: true
+    required: true
+    example_value: xxx-xxx-xxx-xxx
+
+  - name: destinationID
+    description: ID of the destination PU.
+    type: string
+    exposed: true
```
Comment (primalmotion) [inline specs/pingreport.spec]: isn't that required?

File: specs/pingreport.spec [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  - name: destinationNamespace
+    description: Namespace of the destination PU.
+    type: string
+    exposed: true
```
Comment (primalmotion) [inline specs/pingreport.spec]: isn't that required?

File: specs/processingunitrefresh.spec [hunk 3]
```diff
@@ -37,6 +37,14 @@ attributes:
     read_only: true
     autogenerated: true
 
+  - name: pingConfig
```
Comment (primalmotion) [inline specs/processingunitrefresh.spec]: I would prefer dropping the pingConfig sub object and add properties like `pingPorts etc`

File: specs/pingreport.spec (hunk 1 above)
Comment (sibicramesh) [reply specs/pingreport.spec]: no destination can be an external network.

File: specs/pingreport.spec (hunk 2 above)
Comment (sibicramesh) [reply specs/pingreport.spec]: no, destination can be an external network.

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "03b97cf9c989f19574bf74f3eed6aa64f9f40951"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "35784888654ca1099978fa91a3e62c1f6aa3e81c"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "87e4b7d2f8b50ae587f5a8903000d7cb7942730b"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "0ac4dfa5b6ae0a04541aeb9daa91d1b1e8c7df94"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "a0079eddbcc1cac17604796c545f612fbb155ff5"
  }
]
```

File: specs/processingunitrefresh.spec (hunk 3 above)
Comment (sibicramesh) [reply specs/processingunitrefresh.spec]: fixed

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "03b97cf9c989f19574bf74f3eed6aa64f9f40951"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "35784888654ca1099978fa91a3e62c1f6aa3e81c"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "4780ccd592fce9010118054bcb30a7026f927f2f"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "0ac4dfa5b6ae0a04541aeb9daa91d1b1e8c7df94"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "b32602f6994ccd46db0f5775d2a01a191bac4770"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "ping",
    "component": "enforcerd",
    "pr-id": "1526",
    "commit-sha": "03b97cf9c989f19574bf74f3eed6aa64f9f40951"
  },
  {
    "project": "ping",
    "component": "trireme-lib",
    "pr-id": "932",
    "commit-sha": "35784888654ca1099978fa91a3e62c1f6aa3e81c"
  },
  {
    "project": "ping",
    "component": "backend",
    "pr-id": "652",
    "commit-sha": "4780ccd592fce9010118054bcb30a7026f927f2f"
  },
  {
    "project": "ping",
    "component": "gaia",
    "pr-id": "443",
    "commit-sha": "0ac4dfa5b6ae0a04541aeb9daa91d1b1e8c7df94"
  },
  {
    "project": "ping",
    "component": "apoctl",
    "pr-id": "215",
    "commit-sha": "b32602f6994ccd46db0f5775d2a01a191bac4770"
  }
]
```

**Pull request merged**